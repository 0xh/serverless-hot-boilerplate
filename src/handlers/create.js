import serverless from 'serverless-http';
import express from 'express';
import bodyParser from 'body-parser';
import uuid from 'uuid';

import Todo from 'models/todo';

const handler = async (req, res) => {
	try {
		const { text } = req.body;
		const result = await Todo.create({
			id: uuid.v1(),
			text,
			checked: false,
		});
		res.json(result);
	} catch (e) {
		res.json(e);
		throw e;
	}
};

export const app = express().use([bodyParser.json(), handler]);

export default serverless(app);
