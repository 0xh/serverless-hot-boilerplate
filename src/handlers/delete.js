import serverless from 'serverless-http';
import express from 'express';

import Todo from 'models/todo';

export const handler = async (req, res) => {
	try {
		await Todo.delete({
			id: req.params.id,
		});
		res.status(204).end();
	} catch (e) {
		res.json(e);
		throw e;
	}
};

export const app = express().use('/todos/:id', handler);

export default serverless(app);
