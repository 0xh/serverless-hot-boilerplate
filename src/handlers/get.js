import serverless from 'serverless-http';
import express from 'express';

import Todo from 'models/todo';

export const handler = async (req, res) => {
	try {
		const result = await Todo.get({
			id: req.params.id,
		});
		res.json(result);
	} catch (e) {
		res.json(e);
		throw e;
	}
};

export const app = express().use('/todos/:id', handler);

export default serverless(app);
