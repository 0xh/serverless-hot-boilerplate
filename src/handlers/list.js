import serverless from 'serverless-http';
import express from 'express';

import Todo from '../models/todo';

const app = express();

export const handler = async (req, res) => {
	try {
		const result = await Todo.scan().exec();
		res.json(result);
	} catch (e) {
		res.json(e);
		throw e;
	}
};

app.use(handler);

export default serverless(app);
