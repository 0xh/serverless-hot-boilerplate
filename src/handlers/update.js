import serverless from 'serverless-http';
import express from 'express';
import bodyParser from 'body-parser';

import Todo from 'models/todo';

export const handler = async (req, res) => {
	try {
		const result = await Todo.update(
			{
				id: req.params.id,
			},
			req.body,
		);
		res.json(result);
	} catch (e) {
		res.json(e);
		throw e;
	}
};

export const app = express().use('/todos/:id', [bodyParser.json(), handler]);

export default serverless(app);
