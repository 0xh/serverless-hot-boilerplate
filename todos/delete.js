'use strict';

const Todo = require('./model');

module.exports.handler = async (event, context, callback) => {
	try {
		await Todo.delete({
			id: event.pathParameters.id
		});

		const response = {
			statusCode: 200,
			body: JSON.stringify({}),
		};
		callback(null, response);
	} catch(e) {
		console.log(e);
		callback(e);
	}
};
