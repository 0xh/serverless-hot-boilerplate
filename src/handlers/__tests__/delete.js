import request from 'supertest';

import Todo from 'models/todo';
import { app } from 'handlers/delete';

jest.mock('models/todo');

describe('delete.app', () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	test('delete an todo item by id', async () => {
		const mockId = 'xxxx-xxxx';
		const response = await request(app).delete(`/todos/${mockId}`);
		expect(response.status).toBe(204);
		expect(Todo.delete.mock.calls).toMatchSnapshot();
	});
});
