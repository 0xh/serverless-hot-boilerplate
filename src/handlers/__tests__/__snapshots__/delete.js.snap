// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete.app delete an todo item by id 1`] = `
Array [
  Array [
    Object {
      "id": "xxxx-xxxx",
    },
  ],
]
`;
